kolibri/plugins/learn/assets/src/composables/useContentLink.js

```diff
@@ -0,0 +1,66 @@
+import { get } from '@vueuse/core';
+import isEmpty from 'lodash/isEmpty';
+import pick from 'lodash/pick';
+import { computed, getCurrentInstance } from 'kolibri.lib.vueCompositionApi';
+import { PageNames } from '../constants';
+
+export default function useContentLink(store) {
+  // Get store reference from the curent instance
+  // but allow it to be passed in to allow for dependency
+  // injection, primarily for tests.
+  store = store || getCurrentInstance().proxy.$store;
+  const route = computed(() => store.state.route);
+
+  function genContentLink(id, isLeaf = false, repeatParams = false) {
```

Comment:
Looking at the uses of this in practice I think that it might help clarify things to make this take an object so we pass named params wherever it is used. I had to come back and reread the whole function when I got down to where it's passing `true` to the third param. 

This might save someone reading code using this function the trouble of having to read it to remember what that last param is and does

```javascript
genContentLink({id: node.id, is_leaf: node.is_leaf, repeatParams: true});
```

Reply:
Separately, the name `repeatParams` feels a little bit vague here. The comment explaining what it is for is halfway into the function. Maybe something more verbose like `usePreviousQueryParams` or something would be clearer.

Reply:
Perhaps a JSDoc description of the function and its params would be helpful to clarify the purpose up front as well.

Reply:
The other alternative would be to expose two functions, one that generates a link back to this page, and another that generates a link that preserves the query parameter context of the current page.